### DISCUSSION TITLE
Table of Vettel  Laptimes

### DISCUSSION THREAD

**Discussion by Majd-ui** [theOehrly/Fast-F1#110] [Q&A] [answered]
Hello, Im a beginner and Im training myself by putting Vettel's LapTimes in a table.
I managed to get vettel LapTimes in a variable called vet_stint

I just dont know how to make a plot of them in a table or even if its the right thing to do.

My Code:
`Import matplotlib.pyplot as plt`
`import fastf1`
`import fastf1.plotting`
`from fastf1.core import Laps`

`fastf1.Cache.enable_cache('../doc_cache')`

`session = fastf1.get_testing_session(2022, 2, 3)`

`stint = session.load_laps()`

`vet_laps = stint.pick_driver('VET')`
`vet_stint = Laps(vet_laps).sort_values(by='LapTime').reset_index(drop=True)`

`vet = len(vet_stint['LapTime'])`
`amr_color = fastf1.plotting.team_color('AMR')`

I just need the plotting.

**Comment by Majd-ui** [comment]
Update; 
I got as close to what i want with this code :

`import matplotlib.pyplot as plt`
`import pandas as pd`
`import fastf1`
`import fastf1.plotting`
`from fastf1.core import Laps`


`fastf1.Cache.enable_cache('../doc_cache')`

`fastf1.plotting.setup_mpl(mpl_timedelta_support=True, color_scheme=None, misc_mpl_mods=False)`


`stint = session.load_laps().pick_accurate()`


`vet_laps = stint.pick_driver('VET')`
`vet_stint = Laps(vet_laps).sort_values(by='LapNumber')`


`fig, ax =plt.subplots(1,1)`
`column_labels=["LapTime"]`
`ax.axis('tight')`
`ax.axis('off')`
`df=pd.DataFrame(vet_stint.LapTime,columns=column_labels, )`
`ax.table(cellText=df.values,colLabels=df.columns, loc="center")`

`plt.show()`

But Now It is giving the Time in nanoseconds How can i transform it  to M : s : ms
Sorry if it was something basic I have general knowledge in programming specifically web dev, so i dont have much knowledge of python nor data science

**Comment by markste-in** [comment, marked as answer]
maybe try something like this
```
def format(time_value):
    s = pd.Timedelta.total_seconds(time_value)
    return f"{int(s//60)}:{np.round(s%60,3)}"

df["LapTime_formatted"] = df["LapTime"].apply(format)
```
